### DISCUSSION TITLE
Newton Rootfinder: limit max iterations

### DISCUSSION THREAD

**Discussion by ajoardar007** [casadi/casadi#3137] [Q&A]
Hello,

I am trying to limit the maximum number of iterations the newton solver for rootfinder would take to find a root.
I tried to implement the [option](https://casadi.sourceforge.net/v3.0.0-rc1/api/internal/d3/d65/group__rootfinder.html#plugin_Rootfinder_newton) for the Newton method by passing it through opts (Matlab):
`% Options
root_opts = struct();
root_opts.error_on_fail = false;
root_opts.newton.max_iter = 10;
y_disp_EE_root_finder = rootfinder('y_disp_EE_root_finder','newton',y_disp_EE_root_finder_fun,root_opts);`

Is this the correct way of doing it? I was trying to mimic the way IPOPT options are passed on to the nlp function:
`% Options
opts = struct;
opts.ipopt.print_level = 6;
opts.ipopt.acceptable_tol = 1e-2;
opts.ipopt.acceptable_obj_change_tol = 1e-3;

% Create an NLP solver
prob = struct('f', J, 'x', vertcat(w{:}), 'g', vertcat(G{:}));
solver = nlpsol('solver', 'ipopt', prob,opts);`

Please let me know how to pass on the max iteration limit to rootfinder for newton method.

Thanks a lot in advance,
Arnab

**Comment by jgillis** [comment]
Unfortunately, we are not very consistent.
The correct approach is `root_opts.max_iter`.

You may explore options here: https://web.casadi.org/python-api/#rootfinding
